### PR TITLE
Replace template name with package name in module imports

### LINKED ISSUE #40062: [Improvement]: Replace template name with package name in import statements during package creation with templates
### Description

When a user attempts to create a new package using a template pushed to Central, if the template imports any internal modules inside itself, the modules wont be correctly resolved if the user gives a different package name other than the temeplate name and the user will have to change each import statement manually. 
eg:
> The template `egOrg/example_service` has a module `client` which it imports inside it with `import example_service.'client` statement. When a user creates a new package using this as the template with `bal new -t egOrg/example_service eg_service` the import statements inside wont change causing problems with module resolution. 
Changing the import statement to `import eg_service.'client` during the package creation would solve this problem.

### PR DESCRIPTION
**Pull request by ShammiL**
Fixes #40062

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by Dilhasha — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java [hunk 1]
```diff
@@ -285,23 +288,49 @@ public static void deleteDeprecatedMetaFile(Path metaFilePath) {
         }
     }
 
+    public static void replaceTemplateName(Path path, String templateName, String packageName) {
+        Optional<Path> fileName = Optional.ofNullable(path.getFileName());
+        if (fileName.isPresent() && fileName.get().toString().endsWith(BLANG_SOURCE_EXT)) {
```
Comment (gayaldassanayake) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java:293]: Thinking out loud; shouldn't we throw an error if the given bal file is not existing?

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java [hunk 2]
```diff
@@ -567,6 +567,36 @@ public void testNewCommandWithTemplateCentralPullWithVersion() throws IOExceptio
...
+                "distribution = \"" + RepoUtils.getBallerinaShortVersion() + "\"\n\n" +
+                "[build-options]\n" +
+                "observabilityIncluded = true\n";
+        Assert.assertEquals(
+                readFileAsString(packageDir.resolve(ProjectConstants.BALLERINA_TOML)), expectedTomlContent);
+
+        String mainContent = readFileAsString(packageDir.resolve("main.bal"));
+        Assert.assertTrue(mainContent.contains("import central_sample.mod1;"));
+        Assert.assertTrue(Files.exists(packageDir.resolve(ProjectConstants.PACKAGE_MD_FILE_NAME)));
+        Assert.assertTrue(readOutput().contains("Created new package"));
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:597]: Let's execute `bal build` after the package creation to verify there are no diagnostics.

Comment (ShammiL) [reply cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:647]: added

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java [hunk 3]
```diff
@@ -617,6 +617,42 @@ public void testNewCommandWithTemplateCentralPullWithVersion() throws IOExceptio
...
+
+        BuildCommand buildCommand = new BuildCommand(packageDir, printStream, printStream, false);
+        new CommandLine(buildCommand).parseArgs();
+        buildCommand.execute();
```
Comment (azinneera) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/NewCommandTest.java:651]: Let's catch the BLauncher exception here and print the stacktrace.

**Review by azinneera — APPROVED** [review]

**Pull request merged**